libraries/AP_NavEKF3/AP_NavEKF3_core.cpp

```diff
@@ -1313,10 +1308,19 @@ void NavEKF3_core::CovariancePrediction()
         }
     }
 
-    // copy covariances to output
-    CopyCovariances();
+    // covariance matrix is symmetrical, so copy upper half to lower half
+    for (uint8_t row = 1; row <= stateIndexLim; row++) {
+        for (uint8_t column = 0 ; column < row; column++) {
+            P[row][column] = P[column][row] = nextP[column][row];
+        }
+    }
 
-    // constrain diagonals to prevent ill-conditioning
+    // copy variances (diagonals)
+    for (uint8_t i = 0; i <= stateIndexLim; i++) {
+        P[i][i] = nextP[i][i];
```

Comment:
This can be further improved by doing this in the previous loop in one of two ways:

- change `column < row` to `column <= row`
- put this line before or after the inner loop

For either solution `row = 1` needs to be changed to `row = 0`.